dataprep/eda/dtypes.py

```diff
@@ -20,22 +24,152 @@
 DATETIME_DTYPES = DATETIME_NUMPY_DTYPES + DATETIME_PANDAS_DTYPES
 
 
-class DType(Enum):
+class DType:
+    """
+    Root of Type Tree
+    """
+
+
+############## Syntactic DTypes ##############
+class Categorical(DType):
+    """
+    Type Categorical
+    """
+
+
+class Nominal(Categorical):
+    """
+    Type Nominal, Subtype of Categorical
+    """
+
+
+class Ordinal(Categorical):
+    """
+    Type Ordinal, Subtype of Categorical
+    """
+
+
+class Numerical(DType):
+    """
+    Type Numerical
+    """
+
+
+class Continuous(Numerical):
+    """
+    Type Continuous, Subtype of Numerical
+    """
+
+
+class Discrete(Numerical):
     """
-    Possible dtypes for a column, currently we only support categorical and numerical dtypes.
+    Type Discrete, Subtype of Numerical
     """
 
-    Categorical = auto()
-    Numerical = auto()
-    DateTime = auto()
 
+############## Semantic DTypes ##############
 
-def is_categorical(dtype: Any) -> bool:
+
+class DateTime(Numerical):
+    """
+    Type DateTime, Subtype of Numerical
+    """
+
+
+class Text(Nominal):
+    """
+    Type Text, Subtype of Nominal
+    """
+
+
+############## End of the Type Tree ##############
+
+
+def detect_dtype(
+    col: dd.Series, known_dtypes: Optional[Dict[str, Union[DType, str]]] = None
+) -> DType:
+    """
+    Given a column, return its type
+    """
+    if known_dtypes:
+        for col_name, dtype in known_dtypes.items():
```

Comment:
Seems the for loop is not necessary. Could we write `if col_name in known_dtypes` instead?